src/iot/fieldGroupings.spec.js

```diff
@@ -0,0 +1,583 @@
+import omit from 'lodash.omit';
+import FieldGroupings from './fieldGroupings';
+import * as objectUtils from '../utils/objects';
+
+describe('Iot/FieldGroupings', function() {
+  let baseRequest;
+  let baseSdk;
+  let expectedHost;
+
+  beforeEach(function() {
+    this.sandbox = sandbox.create();
+
+    baseRequest = {
+      delete: this.sandbox.stub().resolves(),
+      get: this.sandbox.stub().resolves(),
+      post: this.sandbox.stub().resolves(),
+      put: this.sandbox.stub().resolves()
+    };
+    baseSdk = {
+      config: {
+        audiences: {
+          iot: fixture.build('audience')
+        }
+      }
+    };
+    expectedHost = faker.internet.url();
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  describe('constructor', function() {
+    let fieldGroupings;
+
+    beforeEach(function() {
+      fieldGroupings = new FieldGroupings(baseSdk, baseRequest, expectedHost);
+    });
+
+    it('sets a base url for the class instance', function() {
+      expect(fieldGroupings._baseUrl).to.equal(expectedHost);
+    });
+
+    it('appends the supplied request module to the class instance', function() {
+      expect(fieldGroupings._request).to.equal(baseRequest);
+    });
+
+    it('appends the supplied sdk to the class instance', function() {
+      expect(fieldGroupings._sdk).to.equal(baseSdk);
+    });
+  });
+
+  describe('addField', function() {
+    context('when all required information is supplied', function() {
+      let expectedFieldId;
+      let expectedGroupingField;
+      let expectedGroupingId;
+      let promise;
+      let rawGroupingField;
+      let request;
+      let toCamelCase;
+
+      beforeEach(function() {
+        expectedGroupingField = fixture.build('fieldGroupingField');
+        expectedFieldId = expectedGroupingField.outputFieldId;
+        expectedGroupingId = expectedGroupingField.fieldGroupingId;
+        rawGroupingField = fixture.build('fieldGroupingField', {
```

Comment:
Second parameter is `attributes`. Needs to either be `expectedGroupingField` or `null`.

```
rawGroupingField = fixture.build('fieldGroupingField', expectedGroupingField, {
  fromServer: true
});
```